modules/upnp/upnp_device.cpp

```diff
@@ -35,25 +35,25 @@
 #include <miniupnpc/upnpcommands.h>
 
 String UPNPDevice::query_external_address() const {
-	ERR_FAIL_COND_V(!is_valid_gateway(), "");
+	ERR_FAIL_COND_V_MSG(!is_valid_gateway(), "", "The Internet Gateway Device must be valid.");
 
 	char addr[16];
 	int i = UPNP_GetExternalIPAddress(
 			igd_control_url.utf8().get_data(),
 			igd_service_type.utf8().get_data(),
 			(char *)&addr);
 
-	ERR_FAIL_COND_V(i != UPNPCOMMAND_SUCCESS, "");
+	ERR_FAIL_COND_V_MSG(i != UPNPCOMMAND_SUCCESS, "Couldn't get external IP address.");
```

Comment:
You removed the return value (`""`) so it fails building.

Reply:
> You removed the return value (`""`) so it fails building.

Yes, that was a little slip. I have corrected it and rebased it. I think it should work now.